cms/static/js/programs/views/course_details_view.js

```diff
@@ -0,0 +1,226 @@
+                    courseCodes.push( _.omit(courseData, 'cid') );
+                    this.programModel.patch({ course_codes: courseCodes });
+                }
+            },
+            // Delete this view
+            destroy: function() {
+                Backbone.Validation.unbind(this);
+                this.destroyChildren();
+                this.undelegateEvents();
+                this.removeCourseFromProgram();
+                this.remove();
+            },
+
+            destroyChildren: function() {
+                var runs = this.courseRunViews;
+
+                _.each( runs, function( run ) {
+                    run.removeRun();
+                });
+            },
+
+            // Format data to be passed to the template
+            formatData: function() {
+                var data = $.extend( {}, 
+                    { courseRuns: this.courseRuns.models },
+                    _.omit( this.programModel.toJSON(), 'run_modes'),
+                    this.model.toJSON()
+                );
+
+                return data;
+            },
+
+            removeCourseFromProgram: function() {
+                var courseCodes = this.programModel.get('course_codes'),
+                    key = this.model.get('key'),
+                    name = this.model.get('display_name'),
+                    update = [];
+
+                update = _.reject( courseCodes, function(course) {
+                    return course.key === key && course.display_name === name;
+                });
+
+                this.programModel.patch({ course_codes: update });
+            },
+
+            setCourse: function( event ) {
+                var $form = this.$('.js-course-form'),
+                    title = $form.find('.display-name').val(),
+                    key = $form.find('.course-key').val();
+
+                event.preventDefault();
+
+                this.model.set({
+                    display_name: title,
+                    key: key,
+                    organization: this.programModel.get('organizations')[0]
+                });
+
+                if ( this.model.isValid(true) ) {
+                    this.addCourseToProgram();
+                    this.updateDOM();
+                    this.addCourseRuns();
+                }
+            },
+
+            updateDOM: function() {
+                HtmlUtils.setHtml(this.$el, this.tpl( this.formatData() ) );
+            },
+
+            /**
+             * This is a stripped-down backport of the findIndex function
+             * in underscorejs >= 1.8.0.
+             *
+             * When this app runs within Studio, underscore 1.4.4 is loaded (as
+             * of this writing), so the underscore implementation is not available.
+             *
+             * TODO: use underscore's implementation instead, once Studio has been
+             * updated to use a new-enough version, or the version conflict has
+             * otherwise been resolved.
+             */
```

Comment:
I think Studio is using Underscore 1.8.3 so you should be able to remove the _findIndex backport and just use _.findIndex from Underscore.